### PR TITLE
Allow null as identity for policies

### PR DESCRIPTION
**Pull request by burzum**
Policies can handle non-logged in cases then as well. I've had a discussion with @markstory about this and by removing the requirement of an identity object we can deny or allow access for non-logged cases as well by a policy.

### CODE REVIEW

**Comment by markstory** [comment]
An example use case would be: unauthenticated visitors can view a gallery if it is marked public, otherwise it requires the user to be in the selected group of users.

File: src/AuthorizationServiceInterface.php [hunk 1]
```diff
@@ -25,12 +25,12 @@
...
      * conditional logic around authorization checks.
      *
-     * @param \Authorization\IdentityInterface $user The user to check permissions for.
+     * @param \Authorization\IdentityInterface|null $user The user to check permissions for.
      * @param string $action The action/operation being performed.
      * @param mixed $resource The resource being operated on.
      * @return bool
      */
-    public function can(IdentityInterface $user, $action, $resource);
+    public function can($user, $action, $resource);
```
Comment (dereuromark) [inline src/AuthorizationServiceInterface.php]: Why removing? Can't you just ? nullable it?

Comment (burzum) [reply src/AuthorizationServiceInterface.php]: OK, true, wasn't aware of it, need to get used to the new php features for CakePHP :)

Comment (dereuromark) [reply src/AuthorizationServiceInterface.php]: Just bear in mind that this plugin then needs to add `>=7.1` in https://github.com/cakephp/authorization/blob/master/composer.json#L11

Comment (burzum) [reply src/AuthorizationServiceInterface.php]: @markstory your opinion on that?

Comment (dereuromark) [reply src/AuthorizationServiceInterface.php]: ok. for 4.0 then.

File: src/Controller/Component/AuthorizationComponent.php [hunk 2]
```diff
@@ -143,7 +143,7 @@ protected function getIdentity(ServerRequestInterface $request)
         $identityAttribute = $this->getConfig('identityAttribute');
         $identity = $request->getAttribute($identityAttribute);
         if ($identity === null) {
-            throw new MissingIdentityException([$identityAttribute]);
+            return $identity;
```
Comment (robertpustulka) [inline src/Controller/Component/AuthorizationComponent.php]: Oh no. This makes unauthorized handlers useless and breaks implementations where missing identifiers should be redirected.

Please revert that change.

**Comment by robertpustulka** [comment]
Too bad I have not had a chance to see that before being merged. I don’t agree with this change. How would you redirect missing identities now?

**Comment by robertpustulka** [comment]
@burzum can you update the docs to reflect this change?

**Pull request merged**